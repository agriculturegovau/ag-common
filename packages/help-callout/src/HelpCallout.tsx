import { Callout } from '@ag.ds-next/react/callout';
import { CoreProvider } from '@ag.ds-next/react/core';
import { Text } from '@ag.ds-next/react/text';
import { TextLink, TextLinkExternal } from '@ag.ds-next/react/text-link';

interface HelpCalloutProps {
	hideHelpArticles?: boolean;
	helpHref?: string;
	internal?: boolean;
}

export const HelpCallout = (props: HelpCalloutProps) => {
	const LinkComponent = props?.internal === true ? TextLink : TextLinkExternal;

	return (
		<CoreProvider>
			<Callout
				title={
					props?.hideHelpArticles === true ? 'Need more help?' : 'Need help?'
				}
			>
				{props?.hideHelpArticles === true ? null : (
					<Text>
						Search our{' '}
						<LinkComponent href={props.helpHref ?? '/help'}>Help</LinkComponent>{' '}
						pages
					</Text>
				)}
				<Text>
					Email{' '}
					<TextLink href="mailto:exportservice@aff.gov.au">
						exportservice@aff.gov.au
					</TextLink>
				</Text>
				<Text>
					Call <TextLink href="tel:1800571125">1800&nbsp;571&nbsp;125</TextLink>
					, Monday to Friday, 9 am to 5 pm AEST
				</Text>
			</Callout>
		</CoreProvider>
	);
};
