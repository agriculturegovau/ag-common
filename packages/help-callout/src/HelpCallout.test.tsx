import '@testing-library/jest-dom';
import 'html-validate/jest';
import { render, cleanup } from '../../../test-utils';
import { NeedHelpCallout } from './HelpCallout';

afterEach(cleanup);

describe('NeedHelpCallout', () => {
	it('renders correctly', () => {
		const { container } = render(<NeedHelpCallout />);
		expect(container).toMatchSnapshot();
	});

	it('renders a valid HTML structure', () => {
		const { container } = render(<NeedHelpCallout />);
		expect(container).toHTMLValidate({
			extends: ['html-validate:recommended'],
		});
	});
});
