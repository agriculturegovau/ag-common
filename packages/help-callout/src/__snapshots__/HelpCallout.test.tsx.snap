// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpCallout renders correctly 1`] = `
<div>
  <div
    class="css-1jdu79n-boxStyles-boxStyles"
  >
    <div
      class="css-1baku94-boxStyles-boxStyles-Callout"
    >
      <h2
        class="css-o1mtts-boxStyles-boxStyles"
      >
        Need help?
      </h2>
      <span
        class="css-qrb41g-boxStyles-boxStyles"
      >
        Search our
         
        <a
          class="css-1wvbhdv-TextLink-TextLink"
          href="/help"
          rel="noopener"
          target="_blank"
        >
          Help
          <span
            class="css-5ur8x"
          >
            , opens in a new tab
          </span>
          <svg
            aria-hidden="true"
            class="css-hw4urv-TextLinkExternal"
            clip-rule="evenodd"
            fill-rule="evenodd"
            focusable="false"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 15 21 3M15 3h6M21 9V3M7 3H4.09A1.069 1.069 0 0 0 3 4.037v15.898A1.077 1.077 0 0 0 4.062 21h15.876A1.069 1.069 0 0 0 21 19.934V17"
            />
          </svg>
        </a>
         
        pages
      </span>
      <span
        class="css-qrb41g-boxStyles-boxStyles"
      >
        Email
         
        <a
          class="css-1wvbhdv-TextLink-TextLink"
          href="mailto:exportservice@aff.gov.au"
        >
          exportservice@aff.gov.au
        </a>
      </span>
      <span
        class="css-qrb41g-boxStyles-boxStyles"
      >
        Call 
        <a
          class="css-1wvbhdv-TextLink-TextLink"
          href="tel:1800571125"
        >
          1800 571 125
        </a>
        , Monday to Friday, 9 am to 5 pm AEST
      </span>
    </div>
  </div>
</div>
`;
